security/pfSense-pkg-sudo/files/usr/local/pkg/sudo.inc

```diff
@@ -203,4 +206,73 @@ function sudo_validate_commands(&$input_errors) {
 		}
 	}
 }
+
+function sudo_sync_config() {
+	global $config;
+	if (is_array($config['hasync'])) {
+		$system_carp = $config['hasync'];
+		// XMLRPC sync is currently only supported over connections using the same protocol and port as this system
+		if ($config['system']['webgui']['protocol'] == "http") {
+			$varsyncprotocol = "http";
+			$varsyncport = $config['system']['webgui']['port'] ?: '80';
+		} else {
+			$varsyncprotocol = "https";
+			$varsyncport = $config['system']['webgui']['port'] ?: '443';
+		}
+		sudo_do_xmlrpc_sync($system_carp['synchronizetoip'], $varsyncport, $varsyncprotocol, $system_carp['username'], $system_carp['password'], 100);
+	}
+}
+
+/* Do the actual XMLRPC sync */
+function sudo_do_xmlrpc_sync($sync_to_ip, $port, $protocol, $username, $password, $synctimeout) {
+	global $config;
+
+	/*****************************************************/
+	/* Send over the <sudo> portion of config.xml        */
+	/* $xml will hold the section to sync.               */
+	/*****************************************************/
+	$xml = array();
+	$xml['sudo'] = $config['installedpackages']['sudo'];
+
+	/*************************************************/
+	/* Build a series of commands as a PHP file for  */
+	/* the secondary host to execute to load the new */
+	/* settings.                                     */
+	/*************************************************/
+	$remote_sudo_config_apply = <<<EOD
+<?php
+require_once("/usr/local/pkg/sudo.inc");
+sudo_write_config();
+?>
+EOD;
+
+	/*************************************************/
+	/* First, have target host write the commands    */
+	/* to a PHP file in the /tmp directory.          */
+	/*************************************************/
+	$execcmd = "file_put_contents('/tmp/remote_sudo_config_apply.php', '{$remote_sudo_config_apply}');";
+
+	/*************************************************/
+	/* Now assemble a command to execute the         */
+	/* previously sent PHP file in the background.   */
+	/*************************************************/
+	$execcmd2 = "exec(\"/usr/local/bin/php-cgi -f '/tmp/remote_sudo_config_apply.php' > /dev/null 2>&1 &\");";
+
+	if (pf_version() >= "2.4") {
```

Comment:
No need to check if pfSense is >= 2.4 since the smallest supported version is 2.4.5-p1